contracts/utils/TimestampedHashRegistry.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/utils/cryptography/EIP712.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "./interfaces/ITimestampedHashRegistry.sol";
+
+// TODO: would SelfMulticall be useful?
+contract TimestampedHashRegistry is Ownable, EIP712, ITimestampedHashRegistry {
+    using ECDSA for bytes32;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    mapping(bytes32 => EnumerableSet.AddressSet) private _hashTypeToSigners;
+
+    mapping(bytes32 => SignedHash) public hashTypeToSignedHash;
```

Comment:
Say I'm building Api3Market.sol and I need to fetch the hash of a hash type. I will have to do
```solidity
(bytes32 hash, ) = timestampedHashRegistry.hashTypeToSignedHash(hashType);
```
during which I will also have to pay for `SignedHash.timestamp` to be read from storage even though I don't need it. It's better to keep the hash and the timestamp in different mappings for this reason.

Reply:
Correct but wouldn't that make it more expensive when registering a new hash? (write to `hashTypeToHash` then write to `hashTypeToTimestamp`). It would also require a `view` helper function for reading both values in a single RPC call too I think

Reply:
> Correct but wouldn't that make it more expensive when registering a new hash? (write to hashTypeToHash then write to hashTypeToTimestamp).

There may be an equal cost difference, but it's better to optimize the read than the write because it's expected to be read a lot
> It would also require a view helper function for reading both values in a single RPC call too I think (or callStatic + multicall 🤔 )

You should inherit SelfMulticall anyway

Reply:
> There may be a small difference (much less than an extra storage read), but it's better to optimize the read than the write because it's expected to be read a lot anyway

Sorry if I'm being annoying but isn't this a similar case to [reading a data feed value](https://github.com/api3dao/airnode-protocol-v1/blob/102-TimestampedHashRegistry/contracts/api3-server-v1/Api3ServerV1.sol#L33-L82)? Or is timestamp even useful on this hash registry at all?

Reply:
It isn't because reading a data feed value reads a struct packed in a single bytes32. Here the struct isn't packed, so you'd be paying for reading two bytes32 values even if you won't be needing the timestamp, and the timestamp is much less useful to the user because unless a hash has been overwritten by a new one it still applies, while a data feed value goes stale so the user may want to read the timestamp to check for that.